nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/TransformJSON.java

```diff
@@ -0,0 +1,212 @@
+            .addValidator(new JOLTSpecValidator())
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor JOLT_TRANSFORM = new PropertyDescriptor.Builder()
+            .name("Jolt Transformation")
+            .description("Specifies the Jolt Transformation that should be used with the provided specification.")
+            .required(true)
+            .allowableValues(SHIFTR, CHAINR, DEFAULTR, REMOVR)
+            .defaultValue(CHAINR.getValue())
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("The FlowFile with transformed content will be routed to this relationship")
+            .build();
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("If a FlowFile fails processing for any reason (for example, the FlowFile is not valid JSON), it will be routed to this relationship")
+            .build();
+
+    private List<PropertyDescriptor> properties;
+    private Set<Relationship> relationships;
+    private Transform transform;
+
+
+    @Override
+    protected void init(ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(JOLT_TRANSFORM);
+        properties.add(JOLT_SPEC);
+        this.properties = Collections.unmodifiableList(properties);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, ProcessSession session) throws ProcessException {
+
+        final FlowFile original = session.get();
+        if (original == null) {
+            return;
+        }
+
+        final ProcessorLog logger = getLogger();
+        final StopWatch stopWatch = new StopWatch(true);
+
+        try {
+
+            FlowFile transformed = session.write(original, new StreamCallback() {
+                @Override
+                public void process(final InputStream rawIn, final OutputStream out) throws IOException {
+
+                    try (final InputStream in = new BufferedInputStream(rawIn)) {
+                        Object inputJson = JsonUtils.jsonToObject(in);
+                        Object transformedJson = transform.transform(inputJson);
+                        out.write(JsonUtils.toJsonString(transformedJson).getBytes());
+                    } catch (final Exception e) {
+                        throw new IOException(e);
```

Comment:
What I would also recommend is to use code coverage tools. Not sure what IDE you are using, but if you are using Eclipse, ECL Emma comes with it and by running through it you'll be able to actually see how many of your code was executed. That should help you to either force a scenario in your unit tests for a particular condition or question if that condition is actually valid.